protocol/contracts/farm/facets/SiloFacet/TokenSilo.sol

```diff
@@ -355,6 +366,33 @@ contract TokenSilo is Silo {
             ar.seedsRemoved,
             ar.stalkRemoved
         );
+        return bdvs;
+    }
+
+    function _spendDepositAllowance(
+        address owner,
+        address spender,
+        address token,
+        uint256 amount
+    ) internal virtual {
+        uint256 currentAllowance = depositAllowance(owner, spender, token);
+        if (currentAllowance != type(uint256).max) {
+            require(currentAllowance >= amount, "Silo: insufficient allowance");
+            _approveDeposit(owner, spender, token, currentAllowance - amount);
```

Comment:
Note that this will re-emit the DepositApproval event after every authorized transfer - this isn't required for tokens by the ERC20 standard, but seems to be common behavior.